ocs_ci/ocs/resources/backingstore.py

```diff
@@ -260,33 +266,36 @@ def backingstore_factory(request, cld_mgr, mcg_obj, cloud_uls_factory):
         mcg_obj (MCG): MCG object containing data and utils
             related to MCG
         cloud_uls_factory: Factory for underlying storage creation
+        cluster_context (object): context object in which the bucket will be created.
+            Default is provider context.
 
     Returns:
         func: Factory method - each call to this function creates
             a backingstore
 
     """
-    created_backingstores = []
-
-    cmdMap = {
-        "oc": {
-            "aws": oc_create_aws_backingstore,
-            "gcp": oc_create_google_backingstore,
-            "azure": oc_create_azure_backingstore,
-            "ibmcos": oc_create_ibmcos_backingstore,
-            "rgw": oc_create_rgw_backingstore,
-            "pv": oc_create_pv_backingstore,
-        },
-        "cli": {
-            "aws": cli_create_aws_backingstore,
-            "gcp": cli_create_google_backingstore,
-            "azure": cli_create_azure_backingstore,
-            "ibmcos": cli_create_ibmcos_backingstore,
-            "rgw": cli_create_rgw_backingstore,
-            "pv": cli_create_pv_backingstore,
-            "aws-sts": cli_create_aws_sts_backingstore,
-        },
-    }
+    with cluster_context():
```

Comment:
for the readability and the code execution flow, probably better is to put  with cluster_context(): where we do perform actions, it looks like we  switch to Provider Config to initiate dictionaries. 

Just recommendation